### PR TITLE
Add support for contexts

### PR DESCRIPTION
**Pull request by unmonoqueteclea**
This PR adds support for [Docker Contexts](https://docs.docker.com/engine/context/working-with-contexts/).
This version supports the `ls` and `use` commands. In the future, we could add support for `create` and `inspect`.

Thanks to `Docker Contexts`, you can run docker commands (through `docker.el`) in remote servers.
This is something that we were able to do with `TRAMP` but, based on my experience, this way is much faster.

### CODE REVIEW

File: docker.el [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ;; Author: Philippe Vaucher <philippe.vaucher@gmail.com>
 ;; URL: https://github.com/Silex/docker.el
 ;; Keywords: filename, convenience
-;; Version: 2.2.0
+;; Version: 2.3.0
```
Comment (Silex) [inline docker.el:6]: Can you revert this change? :-)

**Comment by Silex** [comment]
Thanks! This looks like a much better solution than TRAMP indeed.

I'm not sure it's a good idea to add the half-version of it tho, because it's kinda obvious it'll have its own listing like containers/images/volumes do.

Maybe it's better to make it a branch where we at least have the basic listing/dialogs. If you don't have time to do it I'll try to make it happen.

File: docker.el (hunk 1 above)
Comment (unmonoqueteclea) [reply docker.el]: Done :)

**Comment by unmonoqueteclea** [comment]
> I'm not sure it's a good idea to add the half-version of it tho, because it's kinda obvious it'll have its own listing like containers/images/volumes do.
> 
> Maybe it's better to make it a branch where we at least have the basic listing/dialogs. If you don't have time to do it I'll try to make it happen.

If you can wait (one week or two) let me try to do the listing. It's the first emacs package that I contribute to :)
If I am not able to do it, I will let you know.

I am fine with keeping it in my own fork or merging this to a new branch in this repo and closing this PR. Whatever you want.

**Comment by Silex** [comment]
Ok then just add commits to this PR :+1:

Basically you can:
- Copy `docker-volume.el` to `docker-context.el`.
- Replace `volume` by `context`.
- Maybe remove the stuffs about `dangling`.
- Try to make the `create` popup.
- Modify the `docker` popup accordingly so it also displays somes stats.

**Comment by unmonoqueteclea** [comment]
I included support for listings and actions  (remove, inspect, use) for contexts.

The only missing thing is the creation, that I am not sure if we want to include it in the first version (as it is more complex, and this would be the only kind of resource that offers it)

File: docker-core.el [hunk 2]
```diff
@@ -128,7 +128,8 @@
    ("c" (lambda ()(plist-get docker-status-strings :containers)) docker-containers)
    ("i" (lambda ()(plist-get docker-status-strings :images))     docker-images)
    ("n" (lambda ()(plist-get docker-status-strings :networks))   docker-networks)
-   ("v" (lambda ()(plist-get docker-status-strings :volumes))    docker-volumes)]
+   ("v" (lambda ()(plist-get docker-status-strings :volumes))    docker-volumes)
+   ("x" (lambda ()(plist-get docker-status-strings :contexts)) docker-contexts)]
```
Comment (Silex) [inline docker-core.el:132]: Please vertically align `docker-contexts` with the rest :wink:

File: docker-context.el [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+         (data (aio-await (docker-run-docker-async "context" "ls" args (format "--format=\"%s\"" fmt))))
+         (lines (s-split "\n" data t)))
+    (-map (-partial #'docker-utils-parse docker-context-columns) lines)))
+
+(aio-defun docker-context-entries-propertized (&rest args)
+  "Return the propertized docker contexts data for `tabulated-list-entries'."
+  (let ((entries (aio-await (docker-context-entries args)))
+        (active (car (s-split "\n" (aio-await (docker-run-docker-async "context" "show")) t))))
+    (--map-when (string= active (car it)) (docker-context-entry-set-active it) entries)))
+
```
Comment (Silex) [inline docker-context.el:100]: Remove extra line.

File: docker-context.el [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
```
Comment (Silex) [inline docker-context.el:101]: Remove extra line.

File: docker-core.el (hunk 2 above)
Comment (unmonoqueteclea) [reply docker-core.el]: Done!

File: docker-context.el (hunk 4 above)
Comment (unmonoqueteclea) [reply docker-context.el]: Done!

**Comment by Silex** [comment]
Thanks! :+1:
